### ISSUE TITLE
Rebase the cursor when re-ordering layer surfaces

### ISSUE THREAD

**Issue by Limero** [swaywm/sway#3080]
**I first reported this to [slurp](https://github.com/emersion/slurp/issues/15), but this is apparently a sway bug according to @emersion, so the steps below will be from that issue.**

When you run slurp, the cursor will remain regular until you move it, and then, it will turn into the crosshair.

This causes a problem when running slurp and directly doing a selection, the selection will not be highlighted and the resulting selection will have an empty x and y value, like this: 0,0 1144x753

Steps to replicate:
1. Open a terminal and run slurp
2. Don't move the cursor
3. Notice the cursor still being regular, not a crosshair
4. Press the primary mouse button and drag a selection
5. Notice the selection not being highlighted and the result having x/y values set to 0

Sway version: sway version 1.0-beta.1-151-ge9c77970 (Nov  5 2018, branch 'master')

**Comment by emersion** [comment]
We now correctly rebase on map.

We should also rebase on commit if it re-arranges layer surfaces. This requires detecting when layer surfaces need to be moved or resized.

**Comment by Limero** [comment]
Is there more to be done or can this be closed? Cursor updates correctly in slurp after recent changes, so I'm happy :smile:

**Comment by emersion** [comment]
Yes, we should rebase the cursor when a layer surface changes its size.

**Comment by kostadinsh** [comment]
Gentle ping, this is still an issue.

**Comment by emersion** [comment]
Please don't ping in issues. If you want to get this fixed, the only reliable way is to investigate and send a patch.

**Comment by smlx** [comment]
Pretty sure this is fixed in master. At least, I can't reproduce it.

**Comment by kostadinsh** [comment]
> Pretty sure this is fixed in master. At least, I can't reproduce it.

It is somewhat fixed - for some reason the cursor doesn't change when launching `slurp` on one of my monitors, but on the other one it behaves as it should.

The attached files are the output from `WAYLAND_DEBUG=1 slurp` on each of the monitors.

[good-monitor.log](https://github.com/user-attachments/files/28833764/good-monitor.log)

[bad-monitor.log](https://github.com/user-attachments/files/28833766/bad-monitor.log)

**Comment by smlx** [comment]
Thanks for the log. I was able to reproduce the issue.
